leo_docking/leo_docking/states/check_area.py

```diff
@@ -105,24 +90,20 @@ def execute(self, ud):
         self.reset_state()
 
         self.board_id = ud.action_goal.board_id
-        self.node.get_logger().info(f"Waiting for board (id: {self.board_id}) detection.")
+        self.logger.info(f"Waiting for board (id: {self.board_id}) detection.")
 
-        rate = self.node.create_rate(10)
-        time_start = self.node.get_clock().now()
+        start_time = time()
         while not self.board_flag.is_set():
             if self.preempt_requested():
                 self.service_preempt()
                 ud.action_result.result = f"{self.state_log_name}: state preempted."
                 return "preempted"
-            secs = (self.node.get_clock().now() - time_start).nanoseconds//1e9
-            if secs > self.timeout:
-                self.node.get_logger().error(f"Board (id: {self.board_id}) lost. Docking failed.")
-                ud.action_result.result = (
-                    f"{self.state_log_name}: board lost. Docking failed."
-                )
+            if time() - start_time > self.params.timeout:
+                self.logger.error(f"Board (id: {self.board_id}) lost. Docking failed.")
+                ud.action_result.result = f"{self.state_log_name}: board lost. Docking failed."
                 return "board_lost"
 
-            rate.sleep()
+            sleep(0.1)
```

Comment:
Nitpick: If you want to keep the same behavior with the sleep here you might want to use the [rclpy Rate](https://github.com/ros2/rclpy/blob/master/rclpy/rclpy/timer.py#L118) for it. Using that will make the rate of the loop be closer to the correct assuming that the action inside the loop does not take longer than the rate would allow.
```
rate = self.create_rate(10)  # The rate could even be parametrized evetually but let's not do that right now

# The loop
while ....
    ....

   rate.sleep()
```